Fix #1259 by respecting core dir configs

jodal opened the pull request.

jodal commented:
Amended to fix tests.

adamcik commented:
Why is one becoming none, and the other being deprecated? Shouldn't we aim for consistency?

I guess this has to do with the local extensions using the dir as well?

jodal commented:
I removed `local/data_dir` since it feels like a duplicate of `core/data_dir` and `Extension.get_data_dir()`.

`m3u/playlists_dir` I consider to more alike `local/media_dir`, which we still have. It defaults to the extension's data dir, but it would be quite common for users to point it somewhere else entirely, as they probably care more for their playlists and where they are stored (to be used in other apps too?) than they care about where the Mopidy-Local metadata index is stored.

When we split Mopidy-M3U out of Mopidy-Local, @tkem (IIRC) actually made a point of keeping the "playlists_dir" config value name instead of renaming it to `m3u/data_dir`, like I initially did.

adamcik commented:
Right, makes sense. Just wanted to check :-)
